### PR TITLE
Home screen

### PR DESCRIPTION
**Pull request by diegocentenoq**
## 📱 Working on app implementations:

* [x] Upload `Screenrecord/Screenshots` from **Android** implementation.  
* [ ] Upload `Screenrecord/Screenshots` from **iOS** implementation.   

cocktailapp
  
## 🤓 Working on native implementations:  
  
* [ ] Elaborate the change of each specific  
  
## ✚ Adding and Updating Packages? (If applicable)  
  
* [ ] Explain an elaborate the package.  
* [ ] Add to the packages list.  
  
## 💯 Quality  
  
* [x] No new warnings or errors using `fluter analyze`.  
* [x] All private methods start with _ prefix.  
  
**PLEASE IGNORE ANY PR THAT DOES NOT HAVE THIS DESCRIPTION UPDATED**

### CODE REVIEW

File: lib/screens/home/cocktail_app_home_menu_items.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import 'package:boilerplate/styles/images.dart';
+
+enum CocktailAppHomeMenuItems
+{
+  category(name: 'Category'),
+  typeOfGlass(name: 'Type of glass'),
+  ingredient(name: 'Ingredient'),
+  alcoholic(name: 'Alcoholic (or not)');
```
Comment (keylinwu-dnamic) [inline lib/screens/home/cocktail_app_home_menu_items.dart]: Let's please use `Appstrings.current.<name of literal>`
For that, we will need to add the strings in the file `lib/l10n/intl_en.arb`

File: lib/screens/home/cocktail_app_home_menu_items.dart [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  final String name;
+  const CocktailAppHomeMenuItems({required this.name});
+}
+
+extension HomeItemsExt
```
Comment (keylinwu-dnamic) [inline lib/screens/home/cocktail_app_home_menu_items.dart]: Use complete name when naming variables, extension or functions
```suggestion
extension HomeMenuItemsExtension
```

File: lib/styles/images.dart [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CocktailAppImages
+{
+  static const homeScreenBackground = 'assets/images/homescreen_background.png';
+  static const homeScreenTileCategory = 'assets/images/homescreen_tile_category.png';
+  static const homeScreenTileTypeOfGlass = 'assets/images/homescreen_tile_typeofglass.png';
+  static const homeScreenTileIngredient = 'assets/images/homescreen_tile_ingredient.png';
+  static const homeScreenTileAlcoholOrNot = 'assets/images/homescreen_tile_alcoholornot.png';
+}
```
Comment (keylinwu-dnamic) [inline lib/styles/images.dart]: This is great! thanks for adding constants for each of this images paths. A folder asset exists under `lib/constants/asset` Can we add the same contents there. This way we can just centralize absolutely every assets constants in this file

File: lib/screens/home/home_page.dart [hunk 4]
```diff
@@ -49,12 +50,14 @@ class _HomePageState extends ConsumerState<HomePage>
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-      appBar: _buildAppBar(),
+      //appBar: _buildAppBar(),
```
Comment (keylinwu-dnamic) [inline lib/screens/home/home_page.dart]: remove commented code if it is not needed anymore

File: lib/screens/home/home_page.dart [hunk 5]
```diff
@@ -64,6 +67,7 @@ class _HomePageState extends ConsumerState<HomePage>
       backgroundColor: Colors.amber.shade100,
     );
   }
+  */
```
Comment (keylinwu-dnamic) [inline lib/screens/home/home_page.dart]: remove commented code if it is not needed anymore

File: lib/screens/home/widgets/category_main_menu.dart [hunk 6]
```diff
@@ -9,39 +14,95 @@ class CategoryMainMenu extends ConsumerStatefulWidget {
...
+  Container _homeScreenImage()
+  {
+    return Container(
+      decoration: const BoxDecoration(
+          color: CocktailAppColors.aquaGreen,
+          borderRadius: BorderRadius.vertical(bottom: Radius.circular(90)),
+          image: DecorationImage(image: AssetImage(CocktailAppImages.homeScreenBackground), opacity: 0.68, fit: BoxFit.fill)
+      ),
+      width: 393,
+      height: 440,
```
Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: Let's make these `double` variables a constant. Maybe in a `size.dart` static constant file under styles.

File: lib/screens/home/widgets/category_main_menu.dart [hunk 7]
```diff
@@ -9,39 +14,95 @@ class CategoryMainMenu extends ConsumerStatefulWidget {
...
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: const [
+          Text('COCKTAILS APP', textAlign: TextAlign.center, style: CocktailAppFonts.homeTitle),
+          Text('Be your own Bartender', textAlign: TextAlign.center, style: CocktailAppFonts.homeText)
```
Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: String literals should be all localized in `intl_en.arb` file

File: lib/screens/home/widgets/category_main_menu.dart [hunk 8]
```diff
@@ -9,39 +14,95 @@ class CategoryMainMenu extends ConsumerStatefulWidget {
...
+        ],
+      ),
+    );
+  }
+
+  Container _homeScreenMenu(List<Widget> homeScreenMenuItems)
+  {
+    return Container(
+      padding: const EdgeInsets.only(top: 335),
```
Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: Same, let's use the background image size as a constant here. Might be handy in the future

File: lib/screens/home/widgets/category_main_menu.dart [hunk 9]
```diff
@@ -9,39 +14,95 @@ class CategoryMainMenu extends ConsumerStatefulWidget {
...
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children:
+          [
+            Container(
+              height: 100,
+              width: 100,
+              decoration: const BoxDecoration(shape: BoxShape.circle, color: CocktailAppColors.aquaGreen),
+              child: Padding(
+                padding: const EdgeInsets.all(5.0),
```
Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: Let's use `CocktailAppSpacings` constant chere
```suggestion
                padding: const EdgeInsets.all(CocktailAppSpacings.spacingXS),
```

File: lib/screens/home/widgets/category_main_menu.dart [hunk 10]
```diff
@@ -9,39 +14,95 @@ class CategoryMainMenu extends ConsumerStatefulWidget {
...
+                child: CircleAvatar(
+                  backgroundImage: AssetImage(cocktailAppHomeMenuItem.imageFileName),
+                ),
+              ),
+            ),
+            Center(
+              child: Padding(
+                padding: const EdgeInsets.only(top: 10.0, bottom: 0.0),
```
Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: Let's use `CocktailAppSpacings` constant here too

Comment (keylinwu-dnamic) [inline lib/screens/home/widgets/category_main_menu.dart]: Also does bottom padding need to be `0.0`? or can that be just removed?

File: lib/styles/colors.dart [hunk 11]
```diff
@@ -0,0 +1,9 @@
+import 'package:flutter/material.dart';
+
+class CocktailAppColors
+{
+  static const Color white = Color(0xFFFFFFFF);
+  static const Color aquaGreen = Color(0xFF4BFFBE);
+  static const Color lightMint = Color(0xFFF2FFFE);
+  static const Color gray = Color(0xFF323232);
```
Comment (keylinwu-dnamic) [inline lib/styles/colors.dart]: great use of the constants for colors!

File: lib/styles/fonts.dart [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+  static const homeTitle = TextStyle(
+    color: CocktailAppColors.white,
+    fontFamily: 'Arial',
+    fontSize: 40.0,
+    fontWeight: FontWeight.bold,
+    shadows: <Shadow>[
+      Shadow(
+        offset: Offset(3.0, 3.0),
+        blurRadius: 10.0,
+        color: Color.fromARGB(160, 0, 0, 0),
```
Comment (keylinwu-dnamic) [inline lib/styles/fonts.dart]: Can we declare this color before to know which is it, is it gray? or green? o black? then the next developer can totally see it by just reading it.

**Review by keylinwu-dnamic — CHANGES_REQUESTED** [review]
Just some changes required for strings to make them always localized and also some constant missing

File: lib/styles/fonts.dart (hunk 12 above)
Comment (diegocentenoq) [reply lib/styles/fonts.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/styles/colors.dart (hunk 11 above)
Comment (diegocentenoq) [reply lib/styles/colors.dart]: Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/widgets/category_main_menu.dart (hunk 10 above)
Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/widgets/category_main_menu.dart (hunk 9 above)
Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/widgets/category_main_menu.dart (hunk 8 above)
Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/widgets/category_main_menu.dart (hunk 7 above)
Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/widgets/category_main_menu.dart (hunk 6 above)
Comment (diegocentenoq) [reply lib/screens/home/widgets/category_main_menu.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/home_page.dart (hunk 5 above)
Comment (diegocentenoq) [reply lib/screens/home/home_page.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/home_page.dart (hunk 4 above)
Comment (diegocentenoq) [reply lib/screens/home/home_page.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/styles/images.dart (hunk 3 above)
Comment (diegocentenoq) [reply lib/styles/images.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/cocktail_app_home_menu_items.dart (hunk 2 above)
Comment (diegocentenoq) [reply lib/screens/home/cocktail_app_home_menu_items.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/screens/home/cocktail_app_home_menu_items.dart (hunk 1 above)
Comment (diegocentenoq) [reply lib/screens/home/cocktail_app_home_menu_items.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

**Comment by keylinwu-dnamic** [comment]
Just one more thing:
1) Can we remove the green border of the tile and add the shadow on the bottom to match the design?
Screen Shot 2022-06-29 at 17 06 20

File: lib/styles/spacings.dart [hunk 13]
```diff
@@ -0,0 +1,17 @@
+class CocktailAppSpacings
+{
+  // examples
```
Comment (keylinwu-dnamic) [inline lib/styles/spacings.dart]: Remove this comment if it not needed anymore

File: lib/styles/fonts.dart [hunk 14]
```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+import 'package:boilerplate/styles/colors.dart';
+
+class CocktailAppFonts
```
Comment (keylinwu-dnamic) [inline lib/styles/fonts.dart:4]: Good implementations of the fonts 👍

**Review by keylinwu-dnamic — CHANGES_REQUESTED** [review]

File: lib/styles/spacings.dart (hunk 13 above)
Comment (diegocentenoq) [reply lib/styles/spacings.dart]: Suggested changes already pushed. Thank you for your comments and help @keylinwu-dnamic !

File: lib/styles/fonts.dart (hunk 14 above)
Comment (diegocentenoq) [reply lib/styles/fonts.dart:4]: Thank you for your comments and help @keylinwu-dnamic !

**Pull request merged**